### PR TITLE
Added support for cache

### LINKED ISSUE #50: Give us a way to specify cache when the JWTVerifier is created
The Auth0Service uses JWTVerifier from the [auth0-PHP SDK](https://github.com/auth0/auth0-PHP). auth0-PHP allows specifying a [cache](https://github.com/auth0/auth0-PHP#decoding-and-verifying-tokens) which will greatly speed up requests to the Auth0Service. I am currently experience a 3000 millisecond response time for every request that uses the Auth0Service. I manually changed the Auth0Service to use cache and the response time has gone back down to a reasonable 300 milliseconds.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #50. 

I see that you are not running any test on travis. And you barely have any tests... I added some smoke tests. 

I also allowed Symfony 2.8 since it is the same as 3.0.

### CODE REVIEW

**Comment by cocojoe** [comment]
Great thanks, let me look into adding this repo to Travis, Although I see that you're sneaking in 5.5  again as the min requirement 😄 (I can understand why).  Let me check with @glena 
This should also cover the other PRs as well.

**Comment by Nyholm** [comment]
I actually did't sneak it "up" to PHP 5.5. Previous requirements was symfony/symfony:3.0 which require php 5.5.9 ([source](https://github.com/symfony/symfony/blob/3.0/composer.json)). I **lowered** it to php 5.5.0.

**Comment by cocojoe** [comment]
Apologies, I was looking in the other PR history which showed Symfony 2.8 as the requirement.  The ` "php": ">=5.3.3",` seems pretty redundant given the use of Symfony 2.8 | 3.0.

File: composer.json [hunk 1]
```diff
@@ -20,21 +20,24 @@
...
-    "suggest": {
+        "phpunit/phpunit": "^5.1",
+        "nyholm/symfony-bundle-test": "^1.0"
     },
 
     "autoload": {
         "psr-4": {
             "Auth0\\JWTAuthBundle\\": "src/"
         }
     }
```
Comment (glena) [inline composer.json]: missing comma

File: src/Security/Auth0Service.php [hunk 2]
```diff
@@ -23,17 +24,22 @@ class Auth0Service {
...
-    public function __construct($api_secret, $domain, $api_identifier, $authorized_issuer, $secret_base64_encoded, $supported_algs)
+    public function __construct($api_secret, $domain, $api_identifier, $authorized_issuer, $secret_base64_encoded, $supported_algs, CacheHandler $cache = null)
     {
         $this->api_secret = $api_secret;
         $this->domain = $domain;
         $this->api_identifier = $api_identifier;
         $this->authorized_issuer = $authorized_issuer;
         $this->secret_base64_encoded = $secret_base64_encoded;
-        $this->supported_algs = $supported_algs;
+        $this->cache = $cache;
```
Comment (glena) [inline src/Security/Auth0Service.php]: why we did we lose supported algs? (used in decodeJWT)

Comment (Nyholm) [inline src/Security/Auth0Service.php]: Sorry. I did not indent to do that.

File: composer.json (hunk 1 above)
Comment (Nyholm) [inline composer.json]: Thank you!

**Comment by Nyholm** [comment]
Thank you for the review. 

Also, awesome that you enabled Travis =)

**Comment by glena** [comment]
@cocojoe LGTM :)

**Comment by Nyholm** [comment]
Anything else I can do to get this PR merged?

**Comment by cocojoe** [comment]
Thx for reminding me, I'll need to set this repo up for a proper release cycle. Let me sort that out.

**Comment by Nyholm** [comment]
Thank you for merging

**Comment by msznaper** [comment]
Hello, I am using the latest library, but in my symfony it never uses CACHE therefore the credential verification is very slow. How is it activated to work with cache? Thank you so much

**Pull request merged**